pymeasure/instruments/rigol/Rigol_DP932U.py

```diff
@@ -0,0 +1,152 @@
+    """
+    PyMeasure interface for the Rigol DP932U DC Power Supply Unit.
+
+    This class provides methods to control the active channel,
+    voltage, current, output state, and connection mode of the device.
+    It also includes methods to measure voltage and current, reset the device,
+    and check for errors.
+    """
+
+    def __init__(self, adapter, name="Rigol DP932U Power Supply", **kwargs):
+        """
+        Initialize the Rigol DP932U DC Power Supply Unit.
+
+        :param adapter: The communication adapter (e.g., USB or GPIB) to connect to the instrument.
+        :param kwargs: Additional arguments for instrument initialization.
+        """
+
+        if not adapter:
+            raise ValueError("Adapter cannot be None. Provide a valid communication adapter.")
+        kwargs.pop("name", None)
+        super().__init__(adapter, name, **kwargs)
+
+    control_channel = Instrument.control(
+        ":INSTrument:NSELect?",
+        ":INSTrument:NSELect %d",
+        """Control the active channel (1, 2, or 3).
+
+        :param int value: Channel number to set as active.
+        :raises ValueError: If the channel number is outside the valid range [1, 3].
+        """,
+        validator=strict_discrete_set,
+        values=[1, 2, 3],
+    )
+
+    control_voltage = Instrument.control(
+        ":MEASure:VOLTage:DC?",
+        ":SOURce:VOLTage %.3f",
+        """Control the voltage of the selected channel in Volts (0 to 32).
+
+        :param float value: Voltage level to set, within the range [0, 32].
+        :raises ValueError: If the voltage is outside the valid range.
+        """,
+        validator=strict_range,
+        values=[0, 32],
+    )
+
+    control_current = Instrument.control(
+        ":MEASure:CURRent:DC?",
+        ":SOURce:CURRent %.3f",
+        """Control the current of the selected channel in Amps (0 to 3).
+
+        :param float value: Current level to set, within the range [0, 3].
+        :raises ValueError: If the current is outside the valid range.
+        """,
+        validator=strict_range,
+        values=[0, 3],
+    )
+
+    control_output_state = Instrument.control(
+        ":OUTPut:STATe?",
+        ":OUTPut:STATe %s",
+        """Control the output state of the selected channel (ON or OFF).
+
+        :param str value: Output state, either "ON" to enable or "OFF" to disable the output.
+        """,
+        validator=strict_discrete_set,
+        values=["OFF", "ON"],
+        map_values=True,
+    )
+
+    control_connection_mode = Instrument.control(
+        ":OUTPut:PAIR?",
+        ":OUTPut:PAIR %s",
+        """Control the connection mode (OFF, PAR, or SER).
+
+        :param str value: Connection mode, either "OFF", "PAR" (parallel), or "SER" (series).
+        """,
+        validator=strict_discrete_set,
+        values={"OFF": "OFF", "PAR": "PAR", "SER": "SER", "PARALLEL": "PAR", "SERIES": "SER"},
+        map_values=True,
+    )
+
+    @check_error_decorator
```

Comment:
Why don't you just use a pymeasure property "measurement" (like control, but without setting something)?
That will allow to call check_errors as well (see above linked documentation regarding properties).